src/features/formatting/bibtex_internal.rs

```diff
@@ -26,12 +24,30 @@ pub fn format_bibtex_internal(
         indent.push('\t');
     }
 
+    let line_length = {
+        request
+            .context
+            .options
+            .read()
+            .unwrap()
+            .formatter_line_length
+            .map(|value| {
+                if value < 0 {
```

Comment:
should be `<= 0`?

Reply:
Yeah, the fix should be coming soon along with the newlines fix.

Reply:
(If the newline stripping was intentional, that's fine of course.)

Reply:
Hmm... Something is still weird. It uses the default value when I format first; if I undo and format again, it uses the correct value...

(This might be a client issue, though.)

Reply:
Yep, that seems to have done the trick, thank you! 👍🏻 

(sorry for bugging you with all these small things... I'll shut up now ;))

Reply:
No problem, we fixed a lot of (smaller) bugs, now. I think, we can merge this now, can't we?

By the way, what do you think about this feature?
I was thinking about updating the indexing script in order to allow completion of key value options (mainly package imports and other commands that allow `(x)keyval` arguments. The parser is now capable of handling these, so I guess it makes sense to make some use of it. One could even go further and provide warnings using that information if you try to use some unknown key, for example.

Reply:
> No problem, we fixed a lot of (smaller) bugs, now. I think, we can merge this now, can't we?

Yes, I think so, too!

> By the way, what do you think about this feature?
> I was thinking about updating the indexing script in order to allow completion of key value options (mainly package imports and other commands that allow `(x)keyval` arguments. The parser is now capable of handling these, so I guess it makes sense to make some use of it. One could even go further and provide warnings using that information if you try to use some unknown key, for example.

Oh, that's an interesting idea! Yes, I can see how that could be useful. You'd add `[` and `,` as a trigger character for completion, then (assuming the latter isn't already)? Is that something you'd be able to support generally, or do you have to add these explicitly as a `json`?

(If you really want to push the linting, you could also add type information to keys so you could show an error when passing a `bool` for an `integer` value...)

Reply:
> Is that something you'd be able to support generally, or do you have to add these explicitly as a json?

This can be supported generically using the [latex-completion-data](https://github.com/latex-lsp/latex-completion-data) script as a basis. We still have an (incomplete) Rust port around, which makes the script incremental and faster to execute. We can extend this script to detect commands like `\define@key` or `\define@boolkey`, which allows us to provide completion for them without hard-coding every single package (which is basically impossible).